src/Elmish.WPF.Tests/UtilsTests.fs

```diff
@@ -95,3 +95,70 @@ module elmEq =
         let t2 = { t2 with t = t1.t }
         test <@ elmEq t1 t2 = (t1.i = t2.i && t1.s = t2.s) @>
       }
+
+
+module ValueOption =
+
+  module toNullable =
+
+    type ToNullableError =
+      | NullReference
+      | OtherExn of exn
+
+    let tryToNullable<'a> (a: 'a voption) =
+      try
+        ValueOption.toNullable a |> Ok
+      with
+      | :? System.NullReferenceException -> NullReference |> Error
+      | ex -> ex |> OtherExn |> Error
+
+    let testNonNull (ga: Gen<'a>) =
+      Property.check <| property {
+        let! expected = ga
+        test <@ expected = (expected |> ValueSome |> ValueOption.toNullable) @>
+      }
+
+    [<Fact>]
+    let ``toNullable returns ValueSome contents when given ValueSome`` () =
+      testNonNull GenX.auto<obj>
+      testNonNull GenX.auto<string>
+      testNonNull GenX.auto<int>
+
+    let testNullForNullable<'a when 'a : equality> () =
+      test <@ Ok Unchecked.defaultof<'a> = (tryToNullable<'a> ValueNone) @>
+
+    [<Fact>]
+    let ``toNullable returns null when given ValueNone for nullable type`` () =
+      testNullForNullable<obj> ()
+      testNullForNullable<string> ()
+
+    let testNullForNonNullable<'a when 'a : equality> () =
+      test <@ Error NullReference = (tryToNullable<'a> ValueNone) @>
+
+    [<Fact>]
+    let ``toNullable throws NullReferenceException when given ValueNone for nonnullable type`` () =
+      testNullForNonNullable<int> ()
+
```

Comment:
```suggestion
```

Reply:
`testNullForNonNullable` doesn't strictly need removed, but it's unused after the test is removed.

Reply:
We specifically need this to pass